### PR TITLE
Add f401 and f411 blackpill examples

### PR DESCRIPTION
**Pull request by zvecr**
## Description

<https://www.cnx-software.com/2019/12/24/stm32-black-pill-board-features-stm32f4-cortex-m4-mcu-optional-spi-flash/>

* Cheap
* USB C
* Lots of flash

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/onekey/blackpill_f411/rules.mk [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# If you're not setting it here, leave it commented out.
+# It is chip dependent, the correct number can be looked up here (page 175):
+# http://www.st.com/web/en/resource/technical/document/application_note/CD00167594.pdf
+# This also requires a patch to chibios:
+#   <tmk_dir>/tmk_core/tool/chibios/ch-bootloader-jump.patch
+#STM32_BOOTLOADER_ADDRESS = 0x1FFFC800
+
+# Options to pass to dfu-util when flashing
+DFU_ARGS = -d 0483:df11 -a 0 -s 0x08000000:leave
+DFU_SUFFIX_ARGS = -p df11 -v 0483
```
Comment (drashna) [inline keyboards/handwired/onekey/blackpill_f411/rules.mk]:
```suggestion
DFU_SUFFIX_ARGS = -v 0483 -p df11
```

Comment (zvecr) [reply keyboards/handwired/onekey/blackpill_f411/rules.mk]: good catch!

Comment (drashna) [reply keyboards/handwired/onekey/blackpill_f411/rules.mk]: I mean, it's a minor thing, but ... it just bothers me :D

File: drivers/boards/BLACKPILL_STM32_F411/bootloader_defs.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
+/* It is chip dependent, the correct number can be looked up here:
+ * http://www.st.com/web/en/resource/technical/document/application_note/CD00167594.pdf
+ * This also requires a patch to chibios:
+ *  <tmk_dir>/tmk_core/tool/chibios/ch-bootloader-jump.patch
+ */
+#define STM32_BOOTLOADER_ADDRESS 0x1FFF0000
```
Comment (tzarc) [inline drivers/boards/BLACKPILL_STM32_F411/bootloader_defs.h]: I think I'd prefer to remove this file entirely and use it from mcu_selection.mk -- we already have a variable set up for this.

File: drivers/boards/BLACKPILL_STM32_F401/bootloader_defs.h [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
+/* It is chip dependent, the correct number can be looked up here:
+ * http://www.st.com/web/en/resource/technical/document/application_note/CD00167594.pdf
+ * This also requires a patch to chibios:
+ *  <tmk_dir>/tmk_core/tool/chibios/ch-bootloader-jump.patch
+ */
+#define STM32_BOOTLOADER_ADDRESS 0x1FFF0000
```
Comment (tzarc) [inline drivers/boards/BLACKPILL_STM32_F401/bootloader_defs.h]: I think I'd prefer to remove this file entirely and use it from mcu_selection.mk -- we already have a variable set up for this.

**Pull request merged**